p2p/src/authenticated/actors/peer/actor.rs

```diff
@@ -69,17 +71,48 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng> Actor<E> {
     ) -> Result<(), Error> {
         trace!(peer = hex(peer), "sending data",);
 
-        // Send data
         let channel = data.channel;
+        Self::send_message(
+            sender,
+            wire::message::Payload::Data(data),
+            metrics::Message::new_data(peer, channel),
+            sent_messages,
+        )
+        .await?;
+        Ok(())
+    }
+
+    // validate_prioritized_data returns Data after few checks.
+    fn validate_prioritized_data<T>(
+        msg: Option<wire::Data>,
+        contains: T,
```

Comment:
It may be cleaner to just pass in a reference to `rate_limits` rather than using a closure here.